packages/builtin/string.pony

```diff
@@ -139,14 +137,32 @@ actor Main
       _set(0, 0)
     else
       _size = len
+      _alloc = _size + 1
+      _ptr = Pointer[U8]._alloc(_alloc)
+      str._copy_to(_ptr, _alloc)
+    end
 
-      if len == 0 then
-        while str._apply(_size) != 0 do
-          _size = _size + 1
+  new copy_cstring(str: Pointer[U8] box) =>
+    """
+    Create a string by copying a null terminated C string.
+    """
+    if str.is_null() then
+      _size = 0
+      _alloc = 1
+      _ptr = Pointer[U8]._alloc(_alloc)
+      _set(0, 0)
+    else
+      var i: USize = 0
+
+      while true do
+        if str._apply(i) == 0 then
```

Comment:
That piece of code was copied from `from_cstring`, a few lines up.
I've changed both uses to `while str._apply(i) != 0 do`